packages/context/src/inject.ts

```diff
@@ -318,15 +318,15 @@ export namespace inject {
 function resolveAsGetter(
   ctx: Context,
   injection: Readonly<Injection>,
-  session?: ResolutionSession,
+  session: ResolutionSession,
 ) {
   assertTargetIsGetter(injection);
   const bindingSelector = injection.bindingSelector as BindingAddress;
   // We need to clone the session for the getter as it will be resolved later
-  session = ResolutionSession.fork(session);
+  const forkedSession = ResolutionSession.fork(session);
```

Comment:
I see many similarities between `resolveAsGetter` and `resolveAsConfigGetter`, it would be great to share the common bits via a template function. Same for `*View` versions.